mathics/core/expression.py

```diff
@@ -743,7 +755,13 @@ def flatten(
         else:
             return self
 
-    def evaluate(self, evaluation) -> typing.Union["Expression", "Symbol"]:
+    def evaluate(self, evaluation) -> "BaseExpression":
+        """
+        Evaluate until reach a fixed point.
```

Comment:
Personally, I prefer a more expansive style - describing what a fixed point means here. 

But if you don't mind I'll do this at a later date rather than do something now.  

What is here serves is good for now.

Reply:
@rocky, I didn't expand that more because I was not sure how much to say:  Something more verbose could be " `evaluate` applies rules associated to the initial expression until the expression failts to be evaluated again or the resulting expression coincides with the previous one. "
But again, I was not sure how much to put in the docstring and how much in the comments...

Reply:
Sure, I understand. I have made my attempt not in #134. 

The guide I go for in a docstring: the information that a programmer calling the class or method needs to understand.  So here it is what you put in and what you expect to get out from what you put in and how the two are related. 

Specific implementation details like we use a cache to do this, or opinions like I think this code should be rewritten this way are probably better as a comment.